Mathlib/NumberTheory/EulerProduct/Basic.lean

```diff
@@ -0,0 +1,170 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.NumberTheory.SmoothNumbers
+import Mathlib.Analysis.SpecificLimits.Normed
+import Mathlib.Analysis.Normed.Field.InfiniteSum
+
+/-!
+# Euler Products
+
+The main results in this file are `EulerProduct.euler_product`, which says that
+if `f : ℕ → F` is norm-summable, where `F` is a complete normed field and `f` is
+multiplicative on coprime arguments with `f 0 = 0`, then
+`∏ p in primesBelow N, ∑' e : ℕ, f (p^e)` tends to `∑' n, f n` as `N` tends to infinity,
+and `EulerProduct.euler_product_multiplicative`, which states that
+`∏ p in primesBelow N, (1 - f p)⁻¹` tends to `∑' n, f n` as `N` tends to infinity,
+when `f` is completely multiplicative (implemented as `f : ℕ →*₀ F`).
+
+An intermediate step is `EulerProduct.summable_and_hasSum_smoothNumbers_prod_primesBelow_tsum`
+(and its variant `EulerProduct.summable_and_hasSum_smoothNumbers_prod_primesBelow_tsum_geometric`),
+which relates the finite product over primes to the sum of `f n` over `N`-smooth `n`.
+
+## Tags
+
+Euler product
+-/
+
+variable {F : Type*} [NormedField F] [CompleteSpace F]
+
+/-- If `f` is multiplicative and summable, then its values at natural numbers `> 1`
+have norm strictly less than `1`. -/
+lemma Summable.norm_lt_one {f : ℕ →* F} (hsum : Summable f) {p : ℕ} (hp : 1 < p) :
+    ‖f p‖ < 1 := by
+  refine summable_geometric_iff_norm_lt_1.mp ?_
+  simp_rw [← map_pow]
+  exact hsum.comp_injective <| Nat.pow_right_injective hp
+
+
+open scoped Topology
+
+open BigOperators
+
+-- We work with series indexed by the natural numbers
+-- and with terms in a complete normed field `F`.
+variable {F : Type*} [NormedField F] [CompleteSpace F] {f : ℕ → F}
+
+-- We assume that `f` is *multiplicative* in the sense of arithmetic functions,
+-- i.e., multiplicative on coprime elements.
+-- The condition `f 1 = 1` is then equivalent to `f 1 ≠ 0`, but more convenient to use.
+variable (hf₁ : f 1 = 1) (hmul : ∀ {m n}, Nat.Coprime m n → f (m * n) = f m * f n)
+
+lemma Nat.map_prime_pow_mul {p : ℕ} (hp : p.Prime) (e : ℕ) {m : p.smoothNumbers} :
+    f (p ^ e * m) = f (p ^ e) * f m :=
+  hmul <| Nat.Coprime.pow_left _ <| hp.smoothNumbers_coprime <| Subtype.mem m
+
+namespace EulerProduct
+
+open Nat in
```

Comment:
Isn't it simpler to just open this globally throughout the file?

Reply:
Probably. I'll change that.